### PR TITLE
Run CI in GitLab

### PR DESCRIPTION
**Pull request by mitchellhenke**
This PR copies over the checks currently done in CircleCI with the exception of the coverage check. In GitLab, it will report the current coverage, but not fail based on the percentage.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -6,18 +6,129 @@
...
-    - if: '$CI_COMMIT_BRANCH == "main" || $CI_COMMIT_BRANCH == "stages/prod"'
-    - if: '$CI_MERGE_REQUEST_IID || $CI_EXTERNAL_PULL_REQUEST_IID'
+# workflow:
+#   rules:
+#     - if: '$CI_PIPELINE_SOURCE == "schedule"'
+#       when: never
+#     - if: '$CI_PIPELINE_SOURCE == "merge_request_event" || $CI_PIPELINE_SOURCE == "external_pull_request_event"'
+#     - if: '$CI_PIPELINE_SOURCE == "merge_request_event" || $CI_PIPELINE_SOURCE == "external_pull_request_event" || $CI_PIPELINE_SOURCE == "web"'
+#     - if: '$CI_COMMIT_BRANCH == "main" || $CI_COMMIT_BRANCH == "stages/prod"'
+#     - if: '$CI_MERGE_REQUEST_IID || $CI_EXTERNAL_PULL_REQUEST_IID'
```
Comment (zachmargolis) [inline .gitlab-ci.yml:39]: this is temporarily commented out right?

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,8 +1,19 @@
 if ENV['COVERAGE']
   require 'simplecov'
   SimpleCov.start 'rails' do
```
Comment (zachmargolis) [inline spec/spec_helper.rb:3]: ooo this reminds me should we add the RSpec XML formatting output too?

IDP example: https://github.com/18F/identity-idp/pull/7291

**Review by zachmargolis — APPROVED** [review]
LGTM

File: .gitlab-ci.yml (hunk 1 above)
Comment (mitchellhenke) [reply .gitlab-ci.yml]: sorry, yep, brought it back

File: spec/spec_helper.rb (hunk 2 above)
Comment (mitchellhenke) [reply spec/spec_helper.rb:3]: added in 21b228d

**Pull request merged**